### PR TITLE
Updated Bookshelves drop behaviour

### PR DESCRIPTION
**Pull request by Xenoxis**
The dropping behaviour of bookshelves since MC Beta 1.9 Prerelease 3 is dropping 3 books if the tool in hand hasn't silk touch enchantment, otherwise the bookshelf itself is dropped.

Source : 
- https://minecraft.gamepedia.com/Bookshelf#Loot
- https://minecraft.gamepedia.com/Bookshelf#History

### CODE REVIEW

File: src/Blocks/BlockBookShelf.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class cBlockBookShelfHandler :
+	public cBlockHandler
+{
+	public:
+		cBlockBookShelfHandler(BLOCKTYPE a_BlockType)
+			: cBlockHandler(a_BlockType)
+		{
+		}
+
```
Comment (madmaxoft) [inline src/Blocks/BlockBookShelf.h]: There's a leftover tab in here

File: src/Blocks/BlockBookShelf.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+
+class cBlockBookShelfHandler :
+	public cBlockHandler
+{
+	public:
+		cBlockBookShelfHandler(BLOCKTYPE a_BlockType)
+			: cBlockHandler(a_BlockType)
```
Comment (madmaxoft) [inline src/Blocks/BlockBookShelf.h]: Please put the colon at the end of previous line.

**Comment by madmaxoft** [comment]
Other than the minor style issues, this is fine.

**Comment by Xenoxis** [comment]
@madmaxoft Updated again the BlockBookShelf.h again from your request

**Comment by bearbin** [comment]
You need to add yourself to the CONTRIBUTORS file and accept the license, then this will be ready to merge.

File: src/Blocks/BlockBookShelf.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{
+	public:
+		cBlockBookShelfHandler(BLOCKTYPE a_BlockType) :
+			cBlockHandler(a_BlockType)
+		{
+		}
+
+		virtual cItems ConvertToPickups(NIBBLETYPE a_BlockMeta, cBlockEntity * a_BlockEntity, const cEntity * a_Digger, const cItem * a_Tool) override
+		{
+			if ((ItemCategory::IsTool(a_Tool->m_ItemType)) && (ToolHasSilkTouch(a_Tool)))
```
Comment (tigerw) [inline src/Blocks/BlockBookShelf.h]: (minor: redundant parentheses)

Comment (Xenoxis) [reply src/Blocks/BlockBookShelf.h]: I wanted to be respectful of CONTRIBUTING.md : `Add those extra parentheses to conditions, especially in C++`, if isn't needed i can delete them

**Pull request merged**